app/controllers/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+class UsersController < ApplicationController
+  def new
+    if logged_in?
+      redirect_to tasks_path, notice:'既にログインしています'
+    else
+      @user = User.new
+    end
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @tasks = @user.tasks
+    if current_user != @user
+      redirect_to tasks_path, notice:'権限のないユーザーです'
+    end
```

Comment:
elseのないif文は後置ifの記法でコード量を減らせます。

また、redirectさせるかの判定後に@tasksの取得を行った方が必要ない処理が実行されずに済みます。
```suggestion
    redirect_to tasks_path, notice:'権限のないユーザーです' if current_user != @user
    @tasks = @user.tasks
```